src/main/java/uk/gov/hmcts/reform/hmc/service/LinkedHearingGroupServiceImpl.java

```diff
@@ -96,6 +101,57 @@ public void updateLinkHearing(String requestId, HearingLinkGroupRequest hearingL
         validateHearingLinkGroupRequestForUpdate(requestId, hearingLinkGroupRequest);
     }
 
+    @Override
+    public LinkedHearingGroupResponses getLinkedHearingGroupDetails(String requestId) {
+        LinkedGroupDetails linkedGroupDetails =
+            linkedGroupDetailsRepository.getLinkedGroupDetailsByRequestId(requestId);
+        if (linkedGroupDetails != null) {
+            return getLinkedHearingGroupDetails(linkedGroupDetails);
+        } else {
+            throw new LinkedGroupNotFoundException(requestId,INVALID_LINKED_GROUP_REQUEST_ID_DETAILS);
+        }
+    }
+
+    private LinkedHearingGroupResponses getLinkedHearingGroupDetails(LinkedGroupDetails linkedGroupDetails) {
+        LinkedHearingGroupResponses responses = new LinkedHearingGroupResponses();
+        setGroupDetails(linkedGroupDetails,responses);
+        setHearingsInGroup(linkedGroupDetails.getLinkedGroupId(),responses);
+
+        return responses;
+    }
+
+    private void setGroupDetails(LinkedGroupDetails linkedGroupDetails,
+                                 LinkedHearingGroupResponses responses) {
+        GroupDetails groupDetails = new GroupDetails();
+        if (linkedGroupDetails.getRequestName() != null) {
+            groupDetails.setGroupName(linkedGroupDetails.getRequestName());
+        }
+        if (linkedGroupDetails.getReasonForLink() != null) {
+            groupDetails.setGroupReason(linkedGroupDetails.getReasonForLink());
+        }
+        if (linkedGroupDetails.getLinkedComments() != null) {
+            groupDetails.setGroupComments(linkedGroupDetails.getLinkedComments());
+        }
```

Comment:
I don't think these `if`s are required here? We can just set the values without the `if` and it'll have the same outcome as we're not doing anything that could cause a NPE.